### ISSUE TITLE
API: Add endpoint that returns bootcamps that are available for application

### ISSUE THREAD

**Issue by odlbot** [mitodl/bootcamp-ecommerce#447]
**DEPENDS ON: https://github.com/mitodl/bootcamp-ecommerce/issues/468**

[Wireframe](https://projects.invisionapp.com/share/D6WYOGRBHKY#/screens/414528126)

We need an endpoint that fetches a serialized version of every bootcamp that the logged-in user is able to apply for. 

We may add some fields that indicate an overall due date for applications, but for now, we can assume that a user can apply for a bootcamp under these conditions:
1. The bootcamp has a bootcamp run ("klass") that has not started yet (`start_date` < now)
1. The user is not already enrolled in the above bootcamp run

**NOTE: Under current logic, bootcamp runs are only enrollable if we get a certain response back from the fluidreview or SMApply APIs. We need to change this and we need to figure out how.** One option would be to set a flag on the bootcamp/run that indicates that the bootcamp's enrollment is not handled through either of those services, and therefore it is open for application as long as the start date has not passed.

**Comment by pdpinch** [comment]
@noisecapella what's the status of this?

**Comment by noisecapella** [comment]
@gsidebo This is already done, right?

**Comment by gsidebo** [comment]
@noisecapella There is an endpoint for returning all of a user's applications, but I'm pretty sure we don't yet have the endpoint that returns bootcamp runs that a user can apply to
